chart/templates/_init-containers.tpl

```diff
@@ -8,28 +8,12 @@
 {{- end -}}
 {{- end -}}
 
-{{- define "vcluster.k8s.capabilities.version" -}}
-{{/* We need to workaround here for unit tests because Capabilities.KubeVersion.Version is not supported, so we use .Chart.Version */}}
-{{- if hasPrefix "test-" .Chart.Version -}}
-{{- regexFind "^v[0-9]+\\.[0-9]+\\.[0-9]+" (trimPrefix "test-" .Chart.Version) -}}
-{{- else -}}
-{{- regexFind "^v[0-9]+\\.[0-9]+\\.[0-9]+" .Capabilities.KubeVersion.Version -}}
-{{- end -}}
-{{- end -}}
-
 {{/* Bump $defaultTag value whenever k8s version is bumped */}}
 {{- define "vcluster.k8s.image.tag" -}}
-{{- $defaultTag := "v1.32.1" -}}
-{{- if and (not (empty .Values.controlPlane.distro.k8s.version)) (eq .Values.controlPlane.distro.k8s.image.tag $defaultTag) -}}
+{{- if not (empty .Values.controlPlane.distro.k8s.version) -}}
 {{ .Values.controlPlane.distro.k8s.version }}
```

Comment:
With the deprecation of `version` in this release, should we have it try `tag` first and then `version`?

Reply:
Problem is we still need to compare it then to some default as we don't know if it would be set as tag is always set and we cannot determine if the user has set it